### PR TITLE
Add extra salt length check for BCrypt.

### PR DESCRIPTION
**Pull request by Daimanta**

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@Daimanta Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=6969)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security&pullRequestId=6969) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@Daimanta Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security&pullRequestId=6969)!

File: crypto/src/test/java/org/springframework/security/crypto/bcrypt/BCryptTests.java [hunk 1]
```diff
@@ -338,6 +338,11 @@ public void hashpwWorksWithOldRevision() {
 				"$2$05$......................bvpG2UfzdyW/S0ny/4YyEZrmczoJfVm");
 	}
 
+	@Test(expected = IllegalArgumentException.class)
+	public void hashpwFailsWhenSaltisTooShort() {
```
Comment (jzheaux) [inline crypto/src/test/java/org/springframework/security/crypto/bcrypt/BCryptTests.java]: Will you do me the favor of capitalizing "is" here? That way it looks like:

```suggestion
	public void hashpwFailsWhenSaltIsTooShort() {
```

**Comment by jzheaux** [comment]
@Daimanta, are you be able to address the PR feedback?

File: crypto/src/test/java/org/springframework/security/crypto/bcrypt/BCryptTests.java (hunk 1 above)
Comment (Daimanta) [reply crypto/src/test/java/org/springframework/security/crypto/bcrypt/BCryptTests.java]: Changed function name.

**Comment by jzheaux** [comment]
@Daimanta thank you for addressing the inline feedback. 

Will you still be reformatting your comment [as outlined](https://github.com/spring-projects/spring-security/pull/6969#pullrequestreview-247718734)?

**Comment by Daimanta** [comment]
Changed git commit message to comply to the standard.

**Comment by jzheaux** [comment]
Merged this PR via d2248d185b4d0b1758445e5ebafb6422f3385284.

Note that I updated the commit message so that it has a Fixes comment, as outlined [here](https://github.com/spring-projects/spring-security/blob/master/CONTRIBUTING.md#format-commit-messages):

> 4. Include Fixes gh- at the end if this fixes a GitHub issue

**Pull request closed without merging**